### ISSUE TITLE
Image enrichment error on random dockerhub images

### ISSUE THREAD

**Issue by steve-heslouin** [stackrox/stackrox#6273]
Hello Team, 

On stackrox 3.74.3 (installed a month ago). i got random issue on dockerhub , for example, on this image
```
roxctl image scan --image grafana/grafana:latest --endpoint xxxx:443
``` 
it works great!

but on this image
```
roxctl image scan --image grafana/mimir:latest --endpoint xxxx:443
``` 
i got following issue:
```
Error: rpc error: code = Internal desc = image enrichment error: error getting metadata for image: docker.io/grafana/mimir:latest error: getting metadata from registry: "Public DockerHub Public Login": Failed to get the manifest digest : Head "https://registry-1.docker.io/v2/grafana/mimir/manifests/latest": http: non-successful response (status=404 body="")
```

or on this image 
```
roxctl image scan --image pyroscope/pyroscope:latest --endpoint xxxx:443
``` 

the following issue:

```
Error: rpc error: code = Internal desc = image enrichment error: error getting metadata for image: docker.io/pyroscope/pyroscope:latest error: getting metadata from registry: "Public DockerHub Public Login": Get "https://registry-1.docker.io/v2/pyroscope/pyroscope/manifests/sha256:f352b0260521702d1cdd93f6fd35f4234027e056d3b70b3452f9b60335d29574": http: non-successful response (status=404 body="{\"errors\":[{\"code\":\"MANIFEST_UNKNOWN\",\"message\":\"OCI manifest found, but accept header does not support OCI manifests\"}]}\n")
```

Worth mentioning with snyk test or trivy image scan, i do not have any issues, what could explain those failure?

Thanks a lot

**Comment by md2119** [comment]
@steve-heslouin What was your observation retrying `grafana/grafana:latest` after the failures? Did it pass or fail?

**Comment by steve-heslouin** [comment]
It pass, i can run `grafana/grafana:latest` for 10 times in a row without issues , just certain images refuse to work for unknown reasons :(

**Comment by connorgorman** [comment]
@steve-heslouin This is due to a manifest format issue and is fixed in the 4.1 release that should be out in a couple of weeks. See this issue: https://github.com/stackrox/stackrox/issues/5635

**Comment by connorgorman** [comment]
Re-opening as this issue for OCI manifests still remains. Currently investigating

**Comment by connorgorman** [comment]
Merged in a fix, @steve-heslouin Once 4.1 is out, scans should succeed

**Comment by steve-heslouin** [comment]
thanks a lot @connorgorman ;) Looking forward for it ;)
